fix(sql): fix thread and reader leak on invalid filter in WHERE clause

puzpuzpuz opened the pull request:
Fixes #3489

Missing exception handling could lead to thread (and reader) leak due to an infinite loop in `PageFrameSequence#await()` (`dispatchStartFrameIndex` was assigned to `frameCount` in `dispatch()` and never got updated to the proper value).

ideoma commented:
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|1|1|100.00%|

ideoma commented:
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|1|1|100.00%|

puzpuzpuz commented:
@ideoma I've simplified and fixed index initialization in page frame sequence here: d326d60bf78274bc3be6b89e8721d3b0ae1bc1f8

Could take another look?
